src/OpenTelemetry.Instrumentation.Process/ProcessMetrics.cs

```diff
@@ -16,16 +16,31 @@
 
 using System.Diagnostics.Metrics;
 using System.Reflection;
+using Diagnostics = System.Diagnostics;
 
 namespace OpenTelemetry.Instrumentation.Process;
 
 internal class ProcessMetrics
 {
     internal static readonly AssemblyName AssemblyName = typeof(ProcessMetrics).Assembly.GetName();
     internal static readonly Meter MeterInstance = new(AssemblyName.Name, AssemblyName.Version.ToString());
+    private static readonly Diagnostics.Process CurrentProcess = Diagnostics.Process.GetCurrentProcess();
 
     static ProcessMetrics()
     {
+        // TODO: change to ObservableUpDownCounter
+        MeterInstance.CreateObservableGauge(
+            "process.memory.usage",
+            () => CurrentProcess.WorkingSet64,
```

Comment:
It's been awhile since I've studied the difference between `WorkingSet64` and `PrivateMemorySize64`, so I'll admit I'm a bit fuzzy on the semantics of one over the other.

This seems to suggest that PrivateMemorySize64 may be useful as well https://stackoverflow.com/a/1986486/2052722.

Reply:
Ah ha, thanks, yea I agree.

Reply:
I would say all of these are useful (otherwise the NT OS kernel folks wouldn't invent these concepts):

![image](https://user-images.githubusercontent.com/17327289/186786348-f288b3b0-3810-4c73-ab9a-b2766ac52c75.png)

I guess the struggle is do we want to just want to do a .NET specific thing (similar to what we ended up with https://github.com/open-telemetry/opentelemetry-dotnet-contrib/tree/main/src/OpenTelemetry.Instrumentation.Runtime) or try to align with / wait for the spec (or maybe help the spec to move forward). I can also see that the spec might struggle when it comes to some verbose memory management concepts because those could be OS specific (or even specific to certain versions of the NT kernel).

Reply:
Yes, I suspect this would be hard to generalize at the spec level. Maybe something like a `process.dotnet.*` namespace would be useful. Though, yea `process.{os}.{os_distribution}.*` might be more accurate but still hard to generalize across windows, linux, and whatever else.

Reply:
There was an effort a few years back to port these properties to equivalents for linux https://github.com/dotnet/corefx/pull/41122. So at least for .NET and the distros of linux that it supports there should be some level of generalization across OSes.